### PR TITLE
7229 - Add integration to Travis CI

### PR DESCRIPTION
**Pull request by adiroiban**
# Scope

This is a testing PR to check that travis-ci integration works.

For the main discussion see https://twistedmatrix.com/trac/ticket/7229
# Changes

merge @rodrigc branch  and made a few small changes
# How to test

Push changes to the branch associated with the PR and check that travis-ci pick the up and report them to the PR

Travis-ci page should be https://travis-ci.org/twisted/twisted

I don't know what is the link for the build status for a branch

### CODE REVIEW

**Comment by rodrigc** [comment]
I don't understand why those tests are failing.
On this line:
https://travis-ci.org/twisted/twisted/jobs/132773931#L3676

suddenly there are errors about not being able to bind to a socket address.
It looks like something is wrong with the networking setup in the build environment that Travis supplied for this build?

This showed no such errors: https://travis-ci.org/rodrigc/twisted

**Comment by rodrigc** [comment]
If I look at:
https://travis-ci.org/twisted/twisted/jobs/132773931#L11
https://travis-ci.org/rodrigc/twisted/jobs/132701355#L8

There are small differences between the twisted/twisted build environment and the rodrigc/twisted
environment.
- Different provisioning dates on the VM image
- Different git revision on the travis cookbook

I don't know if you can request a specific version for that stuff.
When I set up travis, I pretty much did nothing and just got what they gave me.

**Comment by adiroiban** [comment]
I did the same... I just just click the "enable travis for this repo" button and then turn on
- Build only if .travis.yml is present
- build pushed
- build PR

**Comment by rodrigc** [comment]
So, I took your branch and pushed it to my rodrigc/twisted fork:
https://github.com/rodrigc/twisted/tree/pr_73

I made one small change to enabled building all branches.

Travis built it fine:
https://travis-ci.org/rodrigc/twisted/jobs/132989097

Feel free to create pull requests against rodrigc/twisted if you want to try things out.
I won't merge them.

**Comment by adiroiban** [comment]
I don't think that the tests are failing due to the fact that travis-ci is enalbed only on PR

I have update the config to match your branch... looking for the test results.

I don't know how we can move this forward.

**Comment by rodrigc** [comment]
I've sent mail to support@travis-ci.com to get more help debugging this problem.
Would you be able to merge this branch to trunk, but leave Travis disabled by default on https://github.com/twisted/twisted , to facilitate debugging with the Travis people?

**Comment by rodrigc** [comment]
---------- Forwarded message ----------
From: Dominic Jodoin support@travis-ci.com
Date: Mon, May 30, 2016 at 11:55 AM
Subject: Re: Travis networking problems for https://github.com/twisted/twisted
To: Craig Rodrigues rodrigc@freebsd.org

Hey Craig,

Thanks for reaching out and sorry for this issue.

The different behaviour probably stems from the creation date of both repositories which route their builds on different infrastructure. In brief, repositories created before 2015 are routed on our GCE Precise infrastructure while those created after will be routed on our container-based infrastructure. This is documented here andYou can have more details about our different infrastructures here: https://docs.travis-ci.com/user/ci-environment/#Virtualization-environments.

To route the twisted/twisted builds on the container-based infrastructure (i.e. where your pull request is succeeding) you would need to add the following to your .travis.yml file:

sudo: false

Can you try this out and let me know if it helps? We will follow-up upon your reply.

Thanks in advance and have a great day!

-- Dominic.

How would you rate my reply?
Great    Okay    Not Good

## 

Dominic Jodoin
Customer Support Engineer
support@travis-ci.com

Visit https://www.traviscistatus.com/ for service status and uptime details

Travis CI GmbH, Rigaer Str.8, 10247 Berlin, Germany | GF/CEO: Mathias Meyer, Joshua Kalderimis | Contact: contact@travis-ci.org
Amtsgericht Charlottenburg, Berlin, HRB 140133 B | Umsatzsteuer-ID gemäß §27 a Umsatzsteuergesetz: DE282002648
{#HS:208290677-29798#}  
On Sat, May 28, 2016 at 10:20:49 UTC, Craig Rodrigues rodrigc@freebsd.org wrote:
Hi,

I am helping to set up Travis-CI on the https://github.com/twisted/twisted repository,
and the https://github.com/rodrigc/twisted repository, which is a fork of twisted/twisted.

If I look at:
https://travis-ci.org/twisted/twisted/jobs/132773931#L11
https://travis-ci.org/rodrigc/twisted/jobs/132701355#L8

There are small differences between the twisted/twisted build environment and the rodrigc/twisted
environment.

Different provisioning dates on the VM image
Different git revision on the travis cookbook
I don't know if you can request a specific version for that stuff.
When I set up travis, I pretty much did nothing and just got what Travis gave me.

The twisted/twisted repository fails with networking errors here:
https://travis-ci.org/twisted/twisted/jobs/132773931#L3676

If I take the exact same code, and compile it in the rodrigc/twisted repository,
the build works fine and there are no networking errors:

https://travis-ci.org/rodrigc/twisted/jobs/132989097

Can you shed some light on why there are configuration differences
between the two Travis environments? How can we fix the problem?

Thanks.

## 

Craig

**Comment by adiroiban** [comment]
Thanks. I will try to `sudo: false` ... I knew I read something about sudo and this is why I asked you in a previous review :) hope it will work

**Comment by adiroiban** [comment]
great work @rodrigc  :) many, many thanks!  will merge :)

**Comment by rodrigc** [comment]
How the heck would I have figured out that **sudo: false** would have the side effect of routing builds on Container vs. VM?  Very weird.
Well at least Travis Support responded.

File: CONTRIBUTING [hunk 1]
```diff
@@ -15,9 +15,13 @@ https://twistedmatrix.com/trac/wiki/ContributingToTwistedLabs
 
 Twisted has a Code of Conduct, available at code_of_conduct.md.
 
-**Warning: pull requests are ignored!** File a ticket at:
+**Warning: pull requests are ignored** unless that have an associated ticket.
```
Comment (cyli) [inline CONTRIBUTING]: Grammar nitpick:  "unless **they** have an associated ticket", or "pull requests that don't have an associated ticket are ignored"

**Comment by rodrigc** [comment]
@adiroiban @cyli 
With the tox.ini file as it is now, the Python 2.7 Travis run will fail if it
does not find a topfile in its changeset.  For example, this run failed:

https://travis-ci.org/twisted/twisted/jobs/135160641

because it was triggered via this commit:
https://github.com/twisted/twisted/commit/e56d76fad98d153b4cab70560b0233c93578dce5

which did not have a topfile in its changeset

**Comment by cyli** [comment]
@rodrigc Hm, looks like `_releases.py`'s `CheckTopfileScript` finds all changed files between the merge commit and trunk (there are none) so it thinks there is no topfile.  I think that script should probably be modified to fix that check - it should consider that merge commit to be just the head of trunk.  I can probably fix that in another PR.

**Comment by adiroiban** [comment]
There is this ticket to update the topfile checker https://twistedmatrix.com/trac/ticket/8373

**Pull request merged**